config/types/etcd.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"errors"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/coreos/ignition/config/validate/report"
+)
+
+var (
+	EtcdVersionTooOld      = errors.New("Etcd version specified is not valid (too old)")
+	EtcdMinorVersionTooNew = errors.New("Etcd minor version specified is too new, only options available in the previous minor version will be accepted")
+	EtcdMajorVersionTooNew = errors.New("Etcd version is not valid (too new)")
+	OldestEtcd             = *semver.New("2.3.0")
+)
+
+// Options can be the options for any Etcd version
+type Options interface{}
+
+type etcdCommon Etcd
+
+type EtcdVersion semver.Version
+
+func (e *EtcdVersion) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	t := semver.Version(*e)
+	if err := unmarshal(&t); err != nil {
+		return err
+	}
+	*e = EtcdVersion(t)
+	return nil
+}
+
+func (e EtcdVersion) Validate() report.Report {
+	v := semver.Version(e)
+	switch {
+	case v.LessThan(OldestEtcd):
+		return report.ReportFromError(EtcdVersionTooOld, report.EntryError)
+	case v.Major == 2 && v.Minor > 3:
+	case v.Major == 3 && v.Minor > 1:
```

Comment:
golang case statements don't auto-fallthrough, you'll need to use `fallthrough` explicitly: https://golang.org/ref/spec#Fallthrough_statements